src/charm.py

```diff
@@ -137,8 +137,19 @@ def _configure(self, _):
             container.restart(self._name)
             logger.info("Prometheus (re)started")
 
+        # Provide connection info to any remote write clients.
+        self._set_remote_write()
+
         self.unit.status = ActiveStatus()
 
+    def _set_remote_write(self):
+        """Provide connection info to any remote write clients."""
+        if self.model.relations["ingress"] and self.app.planned_units() == 1:
+            self.remote_write.set_endpoint(address=self._external_hostname, port=self._port)
```

Comment:
Since `self._external_hostname` is currently a config option, does this mean in the future we will need to replace this by a more appropriate solution that will work cross model etc. If so feel free to put a TODO reminder here that is an indication of this technical debt. Or alternatively kindly create a github issue for Prometheus providing context and explaining what kind of changes may be required in the future.

Reply:
I'm not sure I understand. Will the _external_hostname() method not always do what we want here? Could you elaborate on the issue?

Reply:
I may be wrong. But `_external_hostname()` is a bodge because we have no reliable way currently for Juju to tell us what is the externally routable domain name or IP address of any given service or application. Hence `_external_hostname()` was added as a configuration option. Once Juju provides us with this feature then I presume both the remote write library and this method will use whatever Juju provides. Check with @mmanciop who may have better insight into this.

Reply:
I'm not sure juju will ever be able to provide us with this. If it can though, presumably the method `_external_hostname()` can would just change it's logic to use that instead of the config value though. Am I missing something?

Reply:
`_external_hostname()` is just taking a name from a config option and passing it to the Ingress charm, which exposes that name AFAIU to outside the model. All I was asking is this what is intended now and in the future. Perhaps I am going of on a tangent here. Just ignore the rest.